dwave/optimization/model.pyx

```diff
@@ -1764,7 +1764,24 @@ cdef class ArraySymbol(Symbol):
         if exponent == 2:
             from dwave.optimization.symbols import Square  # avoid circular import
             return Square(self)
-        raise ValueError("only squaring is currently supported")
+        # check if exponent is an integer greater than 2
+        elif (isinstance(exponent, int) and exponent > 2) or (isinstance(exponent, float) and exponent.is_integer() and exponent > 2):
```

Comment:
A bit simpler
```python
        elif isinstance(exponent, numbers.Number) and exponent > 0 and int(exponent) == exponent:
```
Also, paired with the below, we can support `exponent=1` so might as well cover that as well.

Reply:
Makes sense, thanks.
I changed numbers.Number to numbers.Real to cover that edge case a little bit more cleanly.
`if isinstance(exponent, numbers.Real) and exponent > 0 and int(exponent) == exponent:`